src/youtube/search.py

```diff
@@ -0,0 +1,38 @@
+import urllib.request
+import re
+import os 
+
+base_url = "https://www.youtube.com/"
+
+def get_videos_ids_from_keywords(scope):
+    context_url = f"results?search_query={scope}"
+    regex_pattern = r"watch\?v=(\S{11})"
+    url = os.path.join(base_url, context_url)
+    html = urllib.request.urlopen(url)
+    video_ids = re.findall(regex_pattern, html.read().decode())
+    return video_ids
+
+
+def get_channel_id_from_video_id(video_id):
+    context_url = f"watch?v={video_id}"
+    regex_pattern = r"<meta itemprop=\"channelId\" content=\"(.*?)\""
+    url = os.path.join(base_url, context_url)
+    html = urllib.request.urlopen(url)
+    channel_id = re.findall(regex_pattern, html.read().decode())
+    return channel_id[0]
+
+
+
+
+def get_channels_ids_from_keywords(scope, video_limit=None):
+    if isinstance(scope, list):
+        youtube_default_sep = "+"
+        scope = youtube_default_sep.join(scope)
+    # video_limit is the max video you want to retry
```

Comment:
@hadjigagny93  i think in this function the better place to put the description of the variable video_limit is in the doc of the function.